File: app/src/main/java/com/aatthhuurr/mobisuas_2411500025/LoginActivity.java
package com.aatthhuurr.mobisuas_2411500025;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.ImageButton;
import android.widget.Toast;
import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;
import com.google.android.material.button.MaterialButton;
import com.google.android.material.checkbox.MaterialCheckBox;
import com.google.android.material.textfield.TextInputEditText;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class LoginActivity extends AppCompatActivity {

    private TextInputEditText edtNoHp, edtPassword;
    private MaterialButton btnLogin;
    private MaterialCheckBox chkRememberMe;
    private MaterialButton btnGoogleLogin;
    private ImageButton btnFingerprint, btnFaceId;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_login);

        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.layoutMainLogin), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });

        // Inisialisasi komponen yang hanya dipakai saja gess
        edtNoHp = findViewById(R.id.edtNoHp);
        edtPassword = findViewById(R.id.edtPassword);
        btnLogin = findViewById(R.id.btnLogin);
        chkRememberMe = findViewById(R.id.chkRememberMe);

        // Aksi Tombol Login Manual
        btnLogin.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String noHp = edtNoHp.getText().toString().trim();
                String password = edtPassword.getText().toString().trim();

                if (noHp.isEmpty()) {
                    edtNoHp.setError("Nomor Handphone wajib diisi!");
                    edtNoHp.requestFocus();
                    return;
                }

                if (password.isEmpty()) {
                    edtPassword.setError("Password wajib diisi!");
                    edtPassword.requestFocus();
                    return;
                }

                // Memanggil API Login ke Server PHP milik temanmu
                ApiClient.getClient().login(noHp, password).enqueue(new Callback<LoginResponse>() {
                    @Override
                    public void onResponse(Call<LoginResponse> call, Response<LoginResponse> response) {
                        if (response.isSuccessful() && response.body() != null) {
                            LoginResponse res = response.body();

                            if (res.getStatus().equalsIgnoreCase("success")) {
                                Toast.makeText(LoginActivity.this, "Login Berhasil!", Toast.LENGTH_SHORT).show();

                                // Pindah ke halaman riwayat gess
                                Intent intent = new Intent(LoginActivity.this, RiwayatActivity.class);
                                intent.putExtra("ID_USER", res.getIdUser());
                                startActivity(intent);
                                finish();
                            } else {
                                Toast.makeText(LoginActivity.this, res.getMessage(), Toast.LENGTH_SHORT).show();
                            }
                        }
                    }

                    @Override
                    public void onFailure(Call<LoginResponse> call, Throwable t) {
                        Toast.makeText(LoginActivity.this, "Gagal koneksi server: " + t.getMessage(), Toast.LENGTH_SHORT).show();
                    }
                });
            }
        });
    }
}

File: app/src/main/java/com/aatthhuurr/mobisuas_2411500025/LoginActivity.kt
package com.aatthhuurr.mobisuas_2411500025

import android.content.Intent
import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.aatthhuurr.mobisuas_2411500025.api.RetrofitClient
import com.aatthhuurr.mobisuas_2411500025.helper.SharedPrefManager
import com.aatthhuurr.mobisuas_2411500025.model.LoginResponse
import com.google.android.material.button.MaterialButton
import com.google.android.material.textfield.TextInputEditText
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class LoginActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        // 1. DIUBAH KE ATAS: Pasang layout XML terlebih dahulu agar findViewById tidak bernilai null saat logout
        setContentView(R.layout.activity_login)

        // 2. Pengecekan Sesi: Jika user sudah login sebelumnya, langsung lempar ke RiwayatActivity
        val prefManager = SharedPrefManager(this)
        if (prefManager.isLoggedIn()) {
            startActivity(Intent(this, RiwayatActivity::class.java))
            finish()
            return // Menghentikan eksekusi kode di bawahnya agar lebih aman
        }

        // 3. Inisialisasi komponen berdasarkan ID di activity_login.xml
        val edtNoHp = findViewById<TextInputEditText>(R.id.edtNoHp)
        val edtPassword = findViewById<TextInputEditText>(R.id.edtPassword)
        val btnLogin = findViewById<MaterialButton>(R.id.btnLogin)

        btnLogin.setOnClickListener {
            val nomorHp = edtNoHp.text.toString().trim()
            val password = edtPassword.text.toString().trim()

            if (nomorHp.isEmpty() || password.isEmpty()) {
                Toast.makeText(this, "Nomor HP dan Password wajib diisi!", Toast.LENGTH_SHORT).show()
            } else {
                prosesLogin(nomorHp, password)
            }
        }
    }

    private fun prosesLogin(nomorHp: String, password: String) {
        RetrofitClient.instance.login(nomorHp, password).enqueue(object : Callback<LoginResponse> {
            override fun onResponse(call: Call<LoginResponse>, response: Response<LoginResponse>) {
                if (response.isSuccessful && response.body() != null) {
                    val loginRes = response.body()!!

                    if (loginRes.status && loginRes.data != null) {
                        // Simpan id_user dan nomor_hp ke SharedPreferences
                        val prefManager = SharedPrefManager(this@LoginActivity)
                        prefManager.saveUser(loginRes.data.id_user, loginRes.data.nomor_hp)

                        Toast.makeText(this@LoginActivity, loginRes.message, Toast.LENGTH_SHORT).show()

                        // Berpindah ke halaman list laporan utama
                        startActivity(Intent(this@LoginActivity, RiwayatActivity::class.java))
                        finish()
                    } else {
                        Toast.makeText(this@LoginActivity, loginRes.message, Toast.LENGTH_SHORT).show()
                    }
                } else {
                    Toast.makeText(this@LoginActivity, "Respon server gagal", Toast.LENGTH_SHORT).show()
                }
            }

            override fun onFailure(call: Call<LoginResponse>, t: Throwable) {
                Toast.makeText(this@LoginActivity, "Gagal terhubung ke server: ${t.message}", Toast.LENGTH_SHORT).show()
            }
        })
    }
}